configure.ac

```diff
@@ -2305,6 +2305,69 @@ then
 fi
 # }}}
 
+# --with-libdpdk {{{
+AC_ARG_WITH(libdpdk, [AS_HELP_STRING([--with-libdpdk@<:@=PREFIX@:>@], [Path to the DPDK build directory.])],
+[
+  if test "x$withval" != "xno" && test "x$withval" != "xyes"
+	then
+		with_dpdk_path="$withval"
+		with_libdpdk="yes"
+	else
+		if test "x$withval" = "xno"
+		then
+			with_libdpdk="no (disabled)"
+		fi
+	fi
+], [with_libdpdk="no"])
+
+if test "x$with_libdpdk" = "xyes"
+then
+	RTE_BUILD="$with_dpdk_path"
+	DPDK_INCLUDE="$RTE_BUILD/include/dpdk"
+	DPDK_LIB_DIR="$RTE_BUILD/lib"
+	DPDK_LIB="-ldpdk"
+	DPDK_EXTRA_LIB="-ldl -lpthread -lrt -lm"
+
+	SAVE_CFLAGS="$CFLAGS"
+	SAVE_LDFLAGS="$LDFLAGS"
+	LDFLAGS="$LDFLAGS -L$DPDK_LIB_DIR"
+	CFLAGS="$CFLAGS -I$DPDK_INCLUDE -fpic"
+	FOUND_DPDK=false
+	SAVE_LIBS="$LIBS"
+	LIBS="$LIBS $DPDK_LIB $DPDK_EXTRA_LIB"
+	AC_CHECK_LIB(dpdk, rte_eal_init,
```

Comment:
You're doing the same check twice, checking for rte_eal_init in libdpdk. The second check is done regardless of the outcome of the first.